noir-projects/aztec-nr/aztec/src/encrypted_logs/payload.nr

```diff
@@ -7,13 +7,71 @@ use dep::std::{embedded_curve_ops::{embedded_curve_add, EmbeddedCurvePoint}, fie
 
 use crate::oracle::unsafe_rand::unsafe_rand;
 
+use crate::event::event_interface::EventInterface;
 use crate::note::note_interface::NoteInterface;
 
 use crate::encrypted_logs::{
     header::EncryptedLogHeader, incoming_body::EncryptedLogIncomingBody,
     outgoing_body::EncryptedLogOutgoingBody
 };
 
+pub fn compute_encrypted_event_log<Event, NB, MB, OB>(
+    contract_address: AztecAddress,
+    randomness: Field,
+    ovsk_app: Field,
+    ovpk: GrumpkinPoint,
+    ivpk: GrumpkinPoint,
+    event: Event
+) -> [u8; OB] where Event: EventInterface<NB, MB> {
+    // @todo Need to draw randomness from the full domain of Fq not only Fr
+    let eph_sk: GrumpkinPrivateKey = fr_to_private_key(unsafe_rand());
+    let eph_pk = eph_sk.derive_public_key();
+
+    // @todo This value needs to be populated!
+    let recipient = AztecAddress::from_field(0);
```

Comment:
I have created #7177 such that we can deal with these. There are two of them here as it was never fully finished 👍